Clarify README (#9)
foundersandcoders/DOM-manipulation-Challenge: README.md

1. Clone this repo.

2. Open `exercise/exercise.js` and follow the instructions.

## What is the DOM?

The DOM is the _Document Object Model_. It is the browser's interpretation of the page once it has been [_parsed_](https://developer.mozilla.org/en-US/docs/Glossary/Parse).

It's something implemented by web browsers rather than provided by the JavaScript language itself.

## DOM elements

An _element_ is the JavaScript representation of something on the page.

> In your browser, if you _right-click_ on a page, you see _inspect element_.

Reviewer: This shows `inspect` rather than `inspect element` in chrome and in the asset image bellow. Perhaps we should consider changing it for consistency?

Author: Yeah good point. I considered taking these screenshots out, since this bit is kind of confusing. I realised that some people may not know about the inspector at this point in the course though

Author: ```suggestion
If you right click on a page you should see an "Inspect" option.
```

Reviewer: We'll be there to help them out :) By the end of the week they'll build the todo list and they'll be extremely used to it. Thanks for editing

</script>
```

`document.getElementsByClassName("myClass")` returns a live [HTML Collection](https://developer.mozilla.org/en-US/docs/Web/API/HTMLCollection) of all elements with the class "myClass";

```html
<li class="menu-item">London
</li>
<li class="menu-item">Nazareth</li>
<script>
  var menuItems = document.getElementsByClassName("menu-item");
  // [li.menu-item, li.menu-item]
</script>
```

> It's important to note that this collection is _live_. If you change one of the elements within it, then try to access the collection again it will be updated. This is different to how most function return values work, so be careful.

Reviewer: Great note. Also, the second workshop of this day is about pure functions. Should we add a comment in brackets that this will be covered in the next wp about pure functions?

Author: I think it might just confuse them to talk about upcoming stuff when they're trying to focus on this

document.querySelector(".myClass"); // logs first element with class name of "myClass"
document.querySelector("p > a.active"); // logs first anchor element with class name of "active" that is within a paragraph
```

`document.querySelectorAll(myCssSelector)` returns a static [NodeList](https://developer.mozilla.org/en-US/docs/Web/API/NodeList) of all elements matching myCssSelector.

```html
<button id="btn-1">Sign up</button>
<button id="btn-2">Sign in</button>
<script>
  var buttons = document.querySelectorAll("button");
  // [button#btn-1, button#btn-2];
</script>
```
It's important to note that this NodeList is _not_ live. Whatever `querySelectorAll` returns will stay the same even if the underlying elements later change. Think of it as a snapshot of the DOM at the time you ran the function.

NodeLists are similar to arrays, but with fewer properties. For example they have a `.forEach()` method, but no `.map()` or `.filter()`. If you need to do more complex operations you can turn them into arrays using `Array.from(myNodeList)`.

```html
<button id="btn-1">Sign up</button>
<button id="btn-2">Sign in</button>
<script>

>   var buttons = document.querySelectorAll("button");

Reviewer: Should we also convert all `var` declarations to es6 and use const, since we're already updating this content?

Author: I considered that too. Will that be confusing if all the other workshops are still using ES5?

Reviewer: It might... Perhaps better to leave it as is, and open an issue to discuss how to update the content to ES6 (unless an issue already exists...) but yeah, it's out of our scope for next week :)

Author: Yeah I think if we do it it should be a new PR

  buttons.forEach(button => {
    console.log(button);

>   }

Reviewer: Missing closing bracket. Should be `});`

Author: Good catch!
```suggestion
  });
```
